jujugui/static/gui/src/app/components/deployment-bar/deployment-bar.js

```diff
@@ -21,17 +21,59 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 YUI.add('deployment-bar', function() {
 
   juju.components.DeploymentBar = React.createClass({
+    previousNotifications: [],
 
-  /**
-    Get the label for the deploy button.
+    /**
+      Get the current state of the deployment bar.
 
-    @method _getDeployButtonLabel
-    @param {Boolean} hasCommits Does the env have commits.
-    @returns {String} the label for the deploy button
-  */
-  _getDeployButtonLabel: function(hasCommits) {
-    return hasCommits ? 'Commit changes' : 'Deploy changes';
-  },
+      @method getInitialState
+      @returns {Object} The current state.
+    */
+    getInitialState: function() {
+      return {
+        latestChangeDescription: null
+      };
+    },
+
+    componentDidMount: function() {
+      this._updateLatestChange(this.props.currentChangeSet);
+    },
+
+    componentWillReceiveProps: function(nextProps) {
```

Comment:
It seems like it's not really possible to test changes that happen as a result of `componentWillReceiveProps` or `componentDidMount`. They don't get fired with the shallow renderer and if you use `renderIntoDocument` it's not really possible to test the output. Happy to hear any suggestions as to how to go about testing these parts.